### PR TITLE
Tasks 6 and 7, Alberto Gómez Jiménez

### PR DESCRIPTION
**Pull request by KairoSagewing5**

### CODE REVIEW

File: Assignment4/KairoSagewing5-22a292/task06.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1xr1sm9yepmy34EGpZpF_nRTsjhMmfMda
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/KairoSagewing5-22a292/task06.py:12]: this is notebook magic, it won't work in python

File: Assignment4/KairoSagewing5-22a292/task06.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+g.add((ns.JaneSmith, vcard.Given, Literal('Jane', datatype = XSD.string)))
+g.add((ns.JaneSmith, vcard.Family, Literal('Smith', datatype = XSD.string)))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+g.add((ns.JohnSmith, vcard.Work, Literal('UPM', datatype = XSD.string)))
```
Comment (dgarijo) [inline Assignment4/KairoSagewing5-22a292/task06.py:72]: You have to add UPM as an entity, not a literal.

Comment (dgarijo) [reply Assignment4/KairoSagewing5-22a292/task06.py:72]: also vcard does not define the property "Work". Define your own

File: Assignment4/KairoSagewing5-22a292/task07.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    UNION
+    {
+      ?Sub RDFS:subClassOf ns:Person.
+      ?Persons RDF:type ?Sub.
+    }
+    UNION
+    {
+      ?Sub RDFS:subClassOf ns:Person.
+        OPTIONAL {?SubSub RDFS:subClassOf ?Sub.}.
+      ?Persons RDF:type ?SubSub.
```
Comment (dgarijo) [inline Assignment4/KairoSagewing5-22a292/task07.py:67]: This does not cover more than 3 levels in the taxonomy. Use * to chain subClassOf properties and retrieve all of them

File: Assignment4/KairoSagewing5-22a292/task07.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    {
+      ?Sub RDFS:subClassOf ns:Person.
+      ?Persons RDF:type ?Sub.
+      ?Persons ?a ?Properties
+    }
+    UNION
+    {
+      ?Sub RDFS:subClassOf ns:Person.
+        OPTIONAL {?SubSub RDFS:subClassOf ?Sub.}.
+      ?Persons RDF:type ?SubSub.
```
Comment (dgarijo) [inline Assignment4/KairoSagewing5-22a292/task07.py:99]: same as above

**Review by dgarijo — CHANGES_REQUESTED** [review]

**Comment by KairoSagewing5** [comment]
The requested changes are done now.

**Comment by dgarijo** [comment]
You are missing exercises 7.1, 7.2 and 7.3 in rdflib

**Comment by KairoSagewing5** [comment]
I don't think so? I see the three Tasks done in the "File Changes" tab. Maybe I'm wrong, but I've tried to clone the repository again just to check, and I see the file [task07.py] with the Tasks done.

**Comment by dgarijo** [comment]
You are solving the exercises using SPARQL, but not in rdflib. If so, indicate the lines, please.

**Comment by KairoSagewing5** [comment]
You were completely right, sorry, I overlooked that part. I've updated the file with the RDFLib part. 
Also, I kept the RDFLib part simple because I could not figure a simple way to iterate queries for subClassOf of chained sub classes (Like with the * in SPARQL) If possible, I would like to hear how to do that.

**Review by dgarijo — APPROVED** [review]

**Comment by dgarijo** [comment]
You can do it with a recursive function.

**Pull request merged**